rust-sdk/whirlpool/src/pool.rs

```diff
@@ -380,3 +380,186 @@ pub async fn fetch_whirlpools_by_token_pair(
 
     Ok(whirlpools)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::tests::{
+        setup_ata_with_amount, setup_mint_with_decimals, setup_whirlpool, RpcContext,
+    };
+    use serial_test::serial;
+    use solana_program_test::tokio;
+    use solana_sdk::signer::Signer;
+
+    struct TestContext {
+        ctx: RpcContext,
+        mint_a: Pubkey,
+        mint_b: Pubkey,
+    }
+
+    impl TestContext {
+        async fn new() -> Result<Self, Box<dyn Error>> {
+            let ctx = RpcContext::new().await;
+            let mint_a = setup_mint_with_decimals(&ctx, 9).await?;
+            let mint_b = setup_mint_with_decimals(&ctx, 9).await?;
+
+            setup_ata_with_amount(&ctx, mint_a, 500_000_000_000).await?;
+            setup_ata_with_amount(&ctx, mint_b, 500_000_000_000).await?;
+
```

Comment:
nit: think if we set up the two pools (splash pool + cl pool) from here we don't have to repeat the logic every time

Reply:
Yep, fixed to use presetup pools